analyses/cell-type-dsrct/exploratory_analysis/01-marker-gene-tumor-classification.Rmd

```diff
@@ -0,0 +1,440 @@
+This would make it hard to identify a cut off to categorize cells as expression or not expressing the marker gene. 
+
+Now we will transform each of the gene expression vectors by generating z-scores. 
+Then we might be able to find a cut off we can use across samples for if marker genes are present in a cell or not. 
+
+```{r}
+umap_df_list_excluded_scaled <- lapply(umap_df_list_excluded, function(x){
+  x   |>
+  dplyr::group_by(gene_symbol) |>
+  # get z-scores for each gene
+  dplyr::mutate(transformed_gene_expression = scale(gene_expression)[, 1]) |>
+  dplyr::ungroup()
+})
+
+```
+
+
+Now we can create the same density plot but looking at z-scores. Interestingly, in some samples, like SPCS000489, we see that ST6GALNAC5 has negative values. Presumably, these are tumor cells but there may be a skew due to the number of tumor cells compared to normal. 
+
+```{r}
+
+for(i in 1:length(umap_df_list_excluded_scaled)) {
+  p <- ggplot(umap_df_list_excluded_scaled[[i]],
+              aes(x = transformed_gene_expression, fill = gene_symbol)) +
+    geom_density() +
+    facet_wrap(vars(gene_symbol), scales = 'free_y') +
+    theme(legend.position = "none") + 
+    ggtitle(names(umap_df_list_excluded_scaled)[i])
+  
+  print(p)
+  
+  #save plots
+  ggsave(
+    filename = paste0(
+      '../plots/marker_distribution_zscaled_',
+      names(umap_df_list)[i],
+      '.png'
+    ),
+    plot = p,
+    width = 6,
+    height = 6,
+    units = 'in',
+    dpi = 150
+  )
+  
+}
+```
+
+It looks like some marker genes have distinct groups of cells with z-score > 0, while other marker genes may not be as informative. 
+
+### Classify tumor cells using marker genes only
+
+Let's try and use the marker gene expression to classify tumor cells. 
+It looks like we could use a cutoff of z-score > 0 to count that cell as a tumor cell. 
+
+We could either count any cell that expresses at least one marker gene > 0 as a tumor cell, or look at the combined expression. 
+Let's start with classifying tumor cells as tumor if any marker gene is present (z-score > 0). 
+This also means the sum of all z-scores > 0. 
+
+Below, we can get the sum of the transformed gene expression of all marker genes and plot in a single UMAP. 
+
+```{r}
+# calculate sum gene expression across all marker genes in list 
+marker_sum_exp <- lapply(umap_df_list_excluded_scaled, function(x){
+  x |>
+  dplyr::group_by(barcodes) |> 
+  dplyr::mutate(sum_exp = sum(transformed_gene_expression, na.rm = T)) |> 
+  dplyr::select(barcodes, UMAP1, UMAP2, sum_exp, cluster) |> 
+  dplyr::distinct()
+})
+
+
+# plot mean gene expression 
+
+for(i in 1:length(marker_sum_exp)) {
+  p <- ggplot(marker_sum_exp[[i]], aes(x = UMAP1, y = UMAP2, color = sum_exp)) +
+  geom_point(size = 0.5, alpha = 0.5) +
+  scale_color_viridis_c() +
+    ggtitle(names(umap_df_list_excluded_scaled)[i])
+  print(p)
+  
+  #save plots
+  ggsave(
+    filename = paste0(
+      '../plots/UMAP_marker_expression_',
```

Comment:
Since we have already defined the module base, we can use that to be a bit more robust to moving the notebook file.
```suggestion
      module_base,
      '/plots/UMAP_marker_expression_',
```